frontend/stores/wallets.ts

```diff
@@ -67,5 +71,22 @@ export const useWalletStore = create<WalletsStore>()((set, get) => ({
     } finally {
       set({ loading: false })
     }
+  },
+  getKeyInfo: (pubKey: string) => {
+    const wallets = get().wallets
+    for (const wallet of wallets) {
+      for (const key of wallet.keys) {
+        if (key.publicKey === pubKey) {
+          return {
+            isOwnKey: true,
+            keyName: key.name
+          }
+        }
+      }
+    }
+    return {
+      isOwnKey: false,
+      keyName: undefined
+    }
```

Comment:
Just return the key, the presence of the key existing is proof that it is owned, also we should just return the whole key rather than a different object the adds no additional information.

Also you can simplify this:
```
return get().wallets.flatMap(w => w.keys).find(({ id }) => id === pubKey)
```